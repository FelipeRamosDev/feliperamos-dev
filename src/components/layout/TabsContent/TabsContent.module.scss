@use '@/style/variables.scss' as *;

.TabsContent {
   .tabContent {
      margin-top: 1rem;
   }

   .tabs {
      border-bottom: 1px solid #333;
   }
}
