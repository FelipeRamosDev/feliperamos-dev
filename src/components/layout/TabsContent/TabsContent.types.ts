export interface TabsContentProps {
   useNewButton?: boolean;
   newContent?: React.ReactNode;
   options: TabOption[];
   className?: string;
   children: React.ReactNode;
}

export interface TabOption {
   label: string;
   value: string;
}
