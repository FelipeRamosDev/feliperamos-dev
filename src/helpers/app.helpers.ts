import { CVData } from '@/types/database.types';
import { defaultLanguage } from '@/app.config';

export function apiURL(path: string, queryParams?: Record<string, string>): string {
   const baseURL = process.env.NEXT_PUBLIC_SERVER_HOST;
   const port = process.env.NEXT_PUBLIC_SERVER_HTTP_PORT;

   if (!baseURL) {
      throw new Error('NEXT_PUBLIC_SERVER_HOST is not defined');
   }

   if (!port) {
      throw new Error('NEXT_PUBLIC_SERVER_HTTP_PORT is not defined');
   }

   const url = new URL(baseURL);
   url.pathname = path;
   url.port = port;

   if (queryParams) {
      Object.entries(queryParams).forEach(([key, value]) => {
         url.searchParams.append(key, value);
      });
   }

   return url.toString();
}

export function cvPDFDownloadLink(cv: CVData, locale: string = defaultLanguage): string {
   const cvId = cv.id;
   const userFullName = cv.user?.name?.replace(/ /g, '_');

   if (!cvId || !userFullName) {
      throw new Error('CV ID or user full name is missing');
   }

   return apiURL(`/cv/${userFullName}-CV_${cvId}_${locale}.pdf`);
}

export function downloadCVPDF(cv: CVData, locale: string): void {
   if (typeof window === 'undefined') {
      return;
   }

   if (!cv) {
      throw new Error('CV data is required to download the PDF');
   }

   const link = cvPDFDownloadLink(cv, locale);
   const a = document.createElement('a');
   const userFullName = cv.user?.name;
   const cvId = cv.id;

   a.href = link;
   a.download = `${userFullName}-${cvId}(${locale}).pdf`;

   document.body.appendChild(a);
   a.click();
   document.body.removeChild(a);
}
