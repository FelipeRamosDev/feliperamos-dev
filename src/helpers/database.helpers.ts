import { FormCheckboxOption, FormSelectOption, FormValues } from "@/hooks/Form/Form.types";
import { TextResources, UserData } from "@/services";
import Ajax from "@/services/Ajax/Ajax";
import { CompanyData, EducationData, ExperienceData, LanguageData, SkillData } from "@/types/database.types";

export const handleExperienceUpdate = async (ajax: Ajax, experience: ExperienceData, values: Partial<ExperienceData>) => {
   if (!values || !Object.keys(values).length) {
      console.warn('No valid updates provided. The "updates" param is empty.');
      return { success: true };
   }

   try {
      const updated = await ajax.post<ExperienceData>('/experience/update', {
         experienceId: experience.id,
         updates: values,
      });

      if (!updated.success) {
         throw updated;
      }

      window.location.reload();
      return { success: true };
   } catch (error) {
      throw error;
   }
};

export async function loadSkillsOptions(ajax: Ajax, textResources: TextResources): Promise<FormSelectOption[]> {
   const { success, data, message } = await ajax.get<SkillData[]>('/skill/query', { params: { language_set: textResources.currentLanguage } });

   if (!success) {
      console.error('Failed to load skills:', message);
      return [];
   }

   return data.map((skill: SkillData) => ({
      value: Number(skill.id),
      label: String(skill.name),
   }));
}

export async function loadCompaniesOptions(ajax: Ajax, textResources: TextResources): Promise<FormSelectOption[]> {
   const { success, data, message } = await ajax.get<CompanyData[]>('/company/query', {
      params: { language_set: textResources.currentLanguage }
   });

   if (!success) {
      console.error('Failed to load companies:', message);
      return [];
   }

   return data.map((company: CompanyData) => ({
      value: Number(company.id),
      label: String(company.company_name)
   }));
}

export async function loadExperiencesListOptions(ajax: Ajax, language_set: string): Promise<FormCheckboxOption[]> {
   try {
      const { data = [] } = await ajax.get<ExperienceData[]>('/experience/query', { params: { language_set } });

      if (!Array.isArray(data)) {
         throw new Error("Failed to load experiences");
      }

      return data.map((item: ExperienceData) => ({
         id: item.id,
         primary: `${item.company?.company_name} (${item.position})`,
         secondary: item.title,
         avatarUrl: item.company?.logo_url
      }));
   } catch (error) {
      console.error("Error loading experiences list options:", error);
      throw error;
   }
}

export async function loadLanguagesOptions(ajax: Ajax, language_set: string): Promise<FormCheckboxOption[]> {
   try {
      const response = await ajax.get<LanguageData[]>('/user/languages', { params: { language_set } });

      if (!response.success) {
         throw new Error("Failed to load languages");
      }

      return response.data.map((item) => ({
         id: item.id,
         primary: item.default_name,
         secondary: item.level,
      }));
   } catch (error) {
      console.error("Error loading languages options:", error);
      throw error;
   }
}

export async function loadEducationsOptions(ajax: Ajax, language_set: string): Promise<FormCheckboxOption[]> {
   try {
      const response = await ajax.get<EducationData[]>('/user/educations', { params: { language_set } });

      if (!response.success) {
         throw new Error("Failed to load educations");
      }

      return response.data.map((item) => ({
         id: item.id,
         primary: item.institution_name,
         secondary: item.field_of_study,
      }));
   } catch (error) {
      console.error("Error loading educations options:", error);
      throw error;
   }
}

export async function updateUserData(ajax: Ajax, data: FormValues): Promise<UserData> {
   try {
      const updatedUser = await ajax.post<UserData>('/user/update', { updates: data });

      if (!updatedUser.success) {
         throw updatedUser;
      }

      window.location.reload();
      return updatedUser.data;
   } catch (error) {
      throw error;
   }
}
