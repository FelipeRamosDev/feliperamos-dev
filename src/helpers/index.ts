export function parseCSS(input: string | string[] = [], defaultValue: string = '') {
   if (typeof input === 'string') {
      input = [ input, defaultValue ].join(' ');
      return input;
   }

   if (Array.isArray(input)) {
      return [ ...input, defaultValue ].join(' ');
   }

   return '';
}
