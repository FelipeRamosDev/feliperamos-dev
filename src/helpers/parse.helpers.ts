export type SizeKeyword = 'xs' | 's' | 'm' | 'l' | 'xl' | 'none';
type ClassInput = string | string[] | undefined | null;

/**
 * Maps padding size keywords to corresponding CSS class names.
 *
 * @param {SizeKeyword} size - Padding size ('xs', 's', 'm', 'l', 'xl', 'none').
 * @returns {string} The CSS class name for the specified padding size, or empty string if 'none' or unrecognized.
 */
export function parsePadding(size: SizeKeyword): string {
   switch (size) {
      case 'xs':
         return 'padding-xs';
      case 's':
         return 'padding-s';
      case 'm':
         return 'padding-m';
      case 'l':
         return 'padding-l';
      case 'xl':
         return 'padding-xl';
      case 'none':
      default:
         return '';
   }
}

/**
 * Maps border-radius size keywords to corresponding CSS class names.
 *
 * @param {SizeKeyword} size - Radius size ('xs', 's', 'm', 'l', 'xl', 'none').
 * @returns {string} The CSS class name for the specified radius size, or empty string if 'none' or unrecognized.
 */
export function parseRadius(size: SizeKeyword): string {
   switch (size) {
      case 'xs':
         return 'radius-xs';
      case 's':
         return 'radius-s';
      case 'm':
         return 'radius-m';
      case 'l':
         return 'radius-l';
      case 'xl':
         return 'radius-xl';
      case 'none':
      default:
         return '';
   }
}

/**
 * Maps elevation size keywords to corresponding CSS class names.
 *
 * @param {SizeKeyword} size - Elevation size ('xs', 's', 'm', 'l', 'xl', 'none').
 * @returns {string} The CSS class name for the specified elevation size, or empty string if 'none' or unrecognized.
 */
export function parseElevation(size: SizeKeyword): string {
   switch (size) {
      case 'xs':
         return 'elevation-xs';
      case 's':
         return 'elevation-s';
      case 'm':
         return 'elevation-m';
      case 'l':
         return 'elevation-l';
      case 'xl':
         return 'elevation-xl';
      case 'none':
      default:
         return '';
   }
}

export function parseBreakpoint(size: SizeKeyword): string {
   switch (size) {
      case 'xs':
         return 'breakpoint-xs';
      case 's':
         return 'breakpoint-s';
      case 'm':
         return 'breakpoint-m';
      case 'l':
         return 'breakpoint-l';
      case 'xl':
         return 'breakpoint-xl';
      case 'none':
      default:
         return '';
   }
}

/**
 * Combines one or two sets of CSS class names into a single string.
 * Accepts strings or arrays of strings and filters out falsy values.
 *
 * @param {ClassInput} classes - First set of class names.
 * @param {ClassInput} merge - Second set of class names to merge.
 * @returns {string} Combined class names separated by spaces.
 */
export function parseCSS(classes?: ClassInput, merge?: ClassInput): string {
   let result: string[] = [];

   if (typeof classes === 'string') {
      result = [...result, ...classes.split(' ')];
   }

   if (Array.isArray(classes)) {
      result = [...result, ...classes];
   }

   if (typeof merge === 'string') {
      result = [...result, ...merge.split(' ')];
   }

   if (Array.isArray(merge)) {
      result = [...result, ...merge];
   }

   return result.filter(item => item).join(' ');
}

/**
 * Formats a numeric value as Brazilian Real currency (R$) with two decimal places.
 *
 * @param {number} value - The numeric value to format.
 * @returns {string} The formatted currency string in 'pt-BR' locale.
 */
export function parseMoney(value: number): string {
   return new Intl.NumberFormat('pt-BR', {
      style: 'currency',
      currency: 'BRL',
      minimumFractionDigits: 2,
      maximumFractionDigits: 2,
   }).format(value);
}

export function parseButtonColorCSS(color?: string): string {
   if (!color || typeof color !== 'string') {
      return '';
   }

   return `button-${color}`;
}

export function onlyNumbers(value: string): string {
   return value.replace(/[^0-9]/g, '');
}
