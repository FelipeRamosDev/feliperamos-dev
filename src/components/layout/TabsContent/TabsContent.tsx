import { parseCSS } from '@/utils/parse';
import { TabsContentProps } from './TabsContent.types';
import { Tab, Tabs } from '@mui/material';
import { useState } from 'react';

export default function TabsContent({ options = [], className, children }: TabsContentProps): React.JSX.Element {
   const [currentTab, setCurrentTab] = useState<number>(0);
   const classes = parseCSS(className, 'TabsContent');

   const handleChange = (_: React.SyntheticEvent, newValue: number) => {
      setCurrentTab(newValue);
   };

   return (
      <div className={classes}>
         <div className="tabs">
            <Tabs value={currentTab} onChange={handleChange}>
               {options.map((option, index) => (
                  <Tab key={index} label={option.label} />
               ))}
            </Tabs>
         </div>

         {children}
      </div>
   );
}
