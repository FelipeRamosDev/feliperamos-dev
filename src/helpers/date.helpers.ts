import { defaultLanguage } from '@/app.config';
import { DateViewStringType } from '@/types/helpers.types';
import dayjs from 'dayjs';

export function dateDifference(startDate: Date, endDate: Date, unit: 'year' | 'month' | 'day'): number {
   const _startDate = dayjs(startDate);
   const _endDate = dayjs(endDate);

   switch (unit) {
      case 'year':
         const years = _endDate.diff(_startDate, 'year');
         return years;
      case 'month':
         const months = _endDate.diff(_startDate, 'month');
         return months;
      case 'day':
         const days = _endDate.diff(_startDate, 'day');
         return days;
      default:
         return 0;
   }
}

export function dataViewString(date: Date, type: DateViewStringType = 'abbMonth-fullYear', locale: string = defaultLanguage): string {
   if (!date) {
      return '---';
   }

   switch (type) {
      case 'locale-standard':
         return dayjs(date).locale(locale).format('LL');
      case 'abbMonth-fullYear':
         return dayjs(date).locale(locale).format('MMM YYYY');
      case 'fullMonth-fullYear':
         return dayjs(date).locale(locale).format('MMMM YYYY');
      case 'fullMonth-abbYear':
         return dayjs(date).locale(locale).format('MMMM YY');
      case 'abbMonth-abbYear':
         return dayjs(date).locale(locale).format('MMM YY');
      default:
         console.warn(`Unknown date view type: ${type}`);
         return '';
   }
};

export function dateDiffYearMonth(startDate: string, endDate: string): { year: number, month: number } {
   const years = dateDifference(new Date(startDate), new Date(endDate), 'year');
   const months = dateDifference(new Date(startDate), new Date(endDate), 'month');
   const restMonths = months % 12;

   return {
      year: years,
      month: restMonths
   };
}

