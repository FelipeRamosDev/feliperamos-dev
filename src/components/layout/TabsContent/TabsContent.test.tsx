import React from 'react';
import { render, screen } from '@testing-library/react';
import TabsContent from './TabsContent';
import { TabsContentProps, TabOption } from './TabsContent.types';

// Mock the parse helpers
jest.mock('@/helpers/parse.helpers', () => ({
   parseCSS: jest.fn(() => 'mocked-parse-css'),
}));

// Mock the SCSS module
jest.mock('./TabsContent.module.scss', () => ({
   TabsContent: 'mocked-tabs-content-class',
   tabs: 'mocked-tabs-class',
   tabContent: 'mocked-tab-content-class',
}));

// Mock Material-UI components
jest.mock('@mui/material', () => ({
   Tabs: ({ children, value, onChange }: { children: React.ReactNode; value?: number; onChange?: () => void }) => (
      <div data-testid="mui-tabs" data-value={value} onClick={onChange}>
         {children}
      </div>
   ),
   Tab: ({ label }: { label?: string }) => (
      <button data-testid="mui-tab" type="button">
         {label}
      </button>
   ),
}));

// Import mocked functions for type checking
import { parseCSS } from '@/helpers/parse.helpers';

const mockParseCSS = parseCSS as jest.MockedFunction<typeof parseCSS>;

describe('TabsContent', () => {
   beforeEach(() => {
      jest.clearAllMocks();
      mockParseCSS.mockReturnValue('mocked-parse-css');
   });

   const defaultOptions: TabOption[] = [
      { label: 'Tab 1', value: 'tab1' },
      { label: 'Tab 2', value: 'tab2' },
      { label: 'Tab 3', value: 'tab3' },
   ];

   const defaultProps: TabsContentProps = {
      options: defaultOptions,
      children: [
         <div key="1">Content 1</div>,
         <div key="2">Content 2</div>,
         <div key="3">Content 3</div>,
      ],
   };

   describe('Basic Rendering', () => {
      it('renders without crashing', () => {
         render(<TabsContent {...defaultProps} />);
         expect(screen.getByTestId('mui-tabs')).toBeInTheDocument();
      });

      it('renders all tab options', () => {
         render(<TabsContent {...defaultProps} />);
         
         expect(screen.getByText('Tab 1')).toBeInTheDocument();
         expect(screen.getByText('Tab 2')).toBeInTheDocument();
         expect(screen.getByText('Tab 3')).toBeInTheDocument();
      });

      it('renders with correct initial tab selected', () => {
         render(<TabsContent {...defaultProps} />);
         
         const tabsContainer = screen.getByTestId('mui-tabs');
         expect(tabsContainer).toHaveAttribute('data-value', '0');
      });

      it('renders first tab content by default', () => {
         render(<TabsContent {...defaultProps} />);
         
         expect(screen.getByText('Content 1')).toBeInTheDocument();
         expect(screen.queryByText('Content 2')).not.toBeInTheDocument();
         expect(screen.queryByText('Content 3')).not.toBeInTheDocument();
      });

      it('renders with div container element', () => {
         const { container } = render(<TabsContent {...defaultProps} />);
         const tabsContent = container.firstChild as HTMLElement;
         
         expect(tabsContent.tagName).toBe('DIV');
         expect(tabsContent).toHaveClass('mocked-parse-css');
      });
   });

   describe('Props Handling', () => {
      describe('options prop', () => {
         it('handles empty options array', () => {
            render(<TabsContent options={[]}><div>No tabs</div></TabsContent>);
            
            expect(screen.queryByTestId('mui-tab')).not.toBeInTheDocument();
            expect(screen.getByText('No tabs')).toBeInTheDocument();
         });

         it('handles single option', () => {
            const singleOption = [{ label: 'Only Tab', value: 'only' }];
            render(
               <TabsContent options={singleOption}>
                  <div>Single content</div>
               </TabsContent>
            );
            
            expect(screen.getByText('Only Tab')).toBeInTheDocument();
            expect(screen.getByText('Single content')).toBeInTheDocument();
         });

         it('handles multiple options correctly', () => {
            const multipleOptions = [
               { label: 'Alpha', value: 'alpha' },
               { label: 'Beta', value: 'beta' },
               { label: 'Gamma', value: 'gamma' },
               { label: 'Delta', value: 'delta' },
            ];
            
            render(
               <TabsContent options={multipleOptions}>
                  <div key="1">Alpha content</div>
                  <div key="2">Beta content</div>
                  <div key="3">Gamma content</div>
                  <div key="4">Delta content</div>
               </TabsContent>
            );
            
            expect(screen.getByText('Alpha')).toBeInTheDocument();
            expect(screen.getByText('Beta')).toBeInTheDocument();
            expect(screen.getByText('Gamma')).toBeInTheDocument();
            expect(screen.getByText('Delta')).toBeInTheDocument();
         });
      });

      describe('className prop', () => {
         it('applies custom className when provided', () => {
            render(<TabsContent {...defaultProps} className="custom-class" />);
            
            expect(mockParseCSS).toHaveBeenCalledWith(
               'custom-class',
               ['TabsContent', 'mocked-tabs-content-class']
            );
         });

         it('does not apply custom className when not provided', () => {
            render(<TabsContent {...defaultProps} />);
            
            expect(mockParseCSS).toHaveBeenCalledWith(
               undefined,
               ['TabsContent', 'mocked-tabs-content-class']
            );
         });
      });

      describe('useNewButton prop', () => {
         it('adds "New" tab when useNewButton is true', () => {
            render(
               <TabsContent 
                  {...defaultProps}
                  useNewButton={true}
                  newContent={<div>New content</div>}
               />
            );
            
            expect(screen.getByText('New')).toBeInTheDocument();
         });

         it('does not add "New" tab when useNewButton is false', () => {
            render(<TabsContent {...defaultProps} useNewButton={false} />);
            
            expect(screen.queryByText('New')).not.toBeInTheDocument();
         });

         it('does not add "New" tab when useNewButton is not provided', () => {
            render(<TabsContent {...defaultProps} />);
            
            expect(screen.queryByText('New')).not.toBeInTheDocument();
         });

         it('shows newContent when "New" tab is selected', () => {
            render(
               <TabsContent 
                  {...defaultProps}
                  useNewButton={true}
                  newContent={<div>New tab content</div>}
               />
            );
            
            // Verify that the "New" tab is added to the options
            expect(screen.getByText('New')).toBeInTheDocument();
            
            // Verify that all tabs including "New" are rendered
            const tabs = screen.getAllByTestId('mui-tab');
            expect(tabs).toHaveLength(4); // 3 default + 1 "New" tab
         });
      });

      describe('newContent prop', () => {
         it('renders newContent when provided with useNewButton', () => {
            const newContentElement = <div>Custom new content</div>;
            render(
               <TabsContent 
                  {...defaultProps}
                  useNewButton={true}
                  newContent={newContentElement}
               />
            );
            
            expect(screen.getByText('New')).toBeInTheDocument();
         });

         it('ignores newContent when useNewButton is false', () => {
            const newContentElement = <div>Should not show</div>;
            render(
               <TabsContent 
                  {...defaultProps}
                  useNewButton={false}
                  newContent={newContentElement}
               />
            );
            
            expect(screen.queryByText('Should not show')).not.toBeInTheDocument();
         });
      });
   });

   describe('Children Handling', () => {
      it('renders single child correctly', () => {
         render(
            <TabsContent options={[{ label: 'Single', value: 'single' }]}>
               <div>Single child content</div>
            </TabsContent>
         );
         
         expect(screen.getByText('Single child content')).toBeInTheDocument();
      });

      it('renders array of children correctly', () => {
         render(<TabsContent {...defaultProps} />);
         
         // Only first child should be visible initially
         expect(screen.getByText('Content 1')).toBeInTheDocument();
         expect(screen.queryByText('Content 2')).not.toBeInTheDocument();
         expect(screen.queryByText('Content 3')).not.toBeInTheDocument();
      });

      it('handles React components as children', () => {
         const CustomComponent = ({ text }: { text: string }) => <span>{text}</span>;
         
         render(
            <TabsContent 
               options={[{ label: 'Component', value: 'comp' }]}>
               <CustomComponent text="Component content" />
            </TabsContent>
         );
         
         expect(screen.getByText('Component content')).toBeInTheDocument();
      });

      it('handles mixed content types as children', () => {
         const mixedChildren = [
            <div key="1">Text content</div>,
            <button key="2">Button content</button>,
            <span key="3">Span content</span>,
         ];
         
         render(
            <TabsContent 
               options={defaultOptions}>
               {mixedChildren}
            </TabsContent>
         );
         
         expect(screen.getByText('Text content')).toBeInTheDocument();
      });
   });

   describe('Tab Switching Logic', () => {
      it('shows correct content when tab is changed', () => {
         render(<TabsContent {...defaultProps} />);
         
         // Initially shows first content
         expect(screen.getByText('Content 1')).toBeInTheDocument();
         expect(screen.queryByText('Content 2')).not.toBeInTheDocument();
         
         // Simulate tab change by re-rendering with different initial state
         // Note: Due to mocking limitations, we test the rendering logic
         expect(screen.getByText('Tab 1')).toBeInTheDocument();
         expect(screen.getByText('Tab 2')).toBeInTheDocument();
         expect(screen.getByText('Tab 3')).toBeInTheDocument();
      });

      it('handles tab change events', () => {
         render(<TabsContent {...defaultProps} />);
         
         const tabsContainer = screen.getByTestId('mui-tabs');
         expect(tabsContainer).toHaveAttribute('data-value', '0');
         
         // Verify that the tabs component receives the onChange handler
         // The mock Tabs component should have an onClick prop that simulates onChange
         expect(tabsContainer).toBeDefined();
         
         // Verify that individual tabs are clickable
         const tabs = screen.getAllByTestId('mui-tab');
         expect(tabs).toHaveLength(3);
         expect(tabs[0]).toHaveTextContent('Tab 1');
      });
   });

   describe('CSS Class Management', () => {
      it('calls parseCSS with correct parameters', () => {
         render(<TabsContent {...defaultProps} className="test-class" />);
         
         expect(mockParseCSS).toHaveBeenCalledWith(
            'test-class',
            ['TabsContent', 'mocked-tabs-content-class']
         );
      });

      it('applies base CSS classes', () => {
         const { container } = render(<TabsContent {...defaultProps} />);
         const tabsContent = container.firstChild as HTMLElement;
         
         expect(tabsContent).toHaveClass('mocked-parse-css');
      });

      it('applies SCSS module classes to child elements', () => {
         const { container } = render(<TabsContent {...defaultProps} />);
         
         expect(container.querySelector('.mocked-tabs-class')).toBeInTheDocument();
         expect(container.querySelector('.mocked-tab-content-class')).toBeInTheDocument();
      });
   });

   describe('State Management', () => {
      it('initializes with first tab selected', () => {
         render(<TabsContent {...defaultProps} />);
         
         const tabsContainer = screen.getByTestId('mui-tabs');
         expect(tabsContainer).toHaveAttribute('data-value', '0');
      });

      it('calculates child count correctly for arrays', () => {
         const multipleChildren = [
            <div key="1">Child 1</div>,
            <div key="2">Child 2</div>,
            <div key="3">Child 3</div>,
            <div key="4">Child 4</div>,
         ];
         
         render(
            <TabsContent 
               options={defaultOptions}
               useNewButton={true}
               newContent={<div>New content</div>}>
               {multipleChildren}
            </TabsContent>
         );
         
         // Should show the "New" tab
         expect(screen.getByText('New')).toBeInTheDocument();
      });

      it('calculates child count correctly for single child', () => {
         render(
            <TabsContent 
               options={[{ label: 'Single', value: 'single' }]}
               useNewButton={true}
               newContent={<div>New content</div>}>
               <div>Single child</div>
            </TabsContent>
         );
         
         expect(screen.getByText('New')).toBeInTheDocument();
      });
   });

   describe('Accessibility', () => {
      it('maintains proper tab structure', () => {
         render(<TabsContent {...defaultProps} />);
         
         expect(screen.getByTestId('mui-tabs')).toBeInTheDocument();
         expect(screen.getAllByTestId('mui-tab')).toHaveLength(3);
      });

      it('preserves tab labels for screen readers', () => {
         render(<TabsContent {...defaultProps} />);
         
         expect(screen.getByText('Tab 1')).toBeInTheDocument();
         expect(screen.getByText('Tab 2')).toBeInTheDocument();
         expect(screen.getByText('Tab 3')).toBeInTheDocument();
      });

      it('does not interfere with child accessibility features', () => {
         render(
            <TabsContent options={[{ label: 'Accessible', value: 'accessible' }]}>
               <div>
                  <button aria-label="Close">×</button>
                  <div role="alert">Important message</div>
               </div>
            </TabsContent>
         );
         
         expect(screen.getByLabelText('Close')).toBeInTheDocument();
         expect(screen.getByRole('alert')).toBeInTheDocument();
      });
   });

   describe('Edge Cases', () => {
      it('handles empty children gracefully', () => {
         render(
            <TabsContent options={[{ label: 'Empty', value: 'empty' }]}>
               {null}
            </TabsContent>
         );
         
         expect(screen.getByText('Empty')).toBeInTheDocument();
      });

      it('handles undefined newContent', () => {
         render(
            <TabsContent 
               {...defaultProps}
               useNewButton={true}
               newContent={undefined}
            />
         );
         
         expect(screen.getByText('New')).toBeInTheDocument();
      });

      it('handles tab options with special characters', () => {
         const specialOptions = [
            { label: 'Tab & Symbol', value: 'special1' },
            { label: 'Tab "Quotes"', value: 'special2' },
            { label: 'Tab <HTML>', value: 'special3' },
         ];
         
         render(
            <TabsContent options={specialOptions}>
               <div key="1">Content 1</div>
               <div key="2">Content 2</div>
               <div key="3">Content 3</div>
            </TabsContent>
         );
         
         expect(screen.getByText('Tab & Symbol')).toBeInTheDocument();
         expect(screen.getByText('Tab "Quotes"')).toBeInTheDocument();
         expect(screen.getByText('Tab <HTML>')).toBeInTheDocument();
      });

      it('handles mismatched options and children count', () => {
         const mismatchedOptions = [
            { label: 'Tab 1', value: 'tab1' },
            { label: 'Tab 2', value: 'tab2' },
         ];
         
         render(
            <TabsContent options={mismatchedOptions}>
               <div key="1">Content 1</div>
               <div key="2">Content 2</div>
               <div key="3">Content 3</div>
            </TabsContent>
         );
         
         expect(screen.getByText('Tab 1')).toBeInTheDocument();
         expect(screen.getByText('Tab 2')).toBeInTheDocument();
         expect(screen.getByText('Content 1')).toBeInTheDocument();
      });

      it('handles complex nested content', () => {
         const complexContent = [
            <div key="1">
               <header>
                  <h2>Tab 1 Header</h2>
               </header>
               <main>
                  <p>Tab 1 content</p>
               </main>
            </div>,
            <div key="2">
               <form>
                  <input placeholder="Form input" />
                  <button type="submit">Submit</button>
               </form>
            </div>,
         ];
         
         render(
            <TabsContent 
               options={[
                  { label: 'Complex 1', value: 'complex1' },
                  { label: 'Complex 2', value: 'complex2' },
               ]}>
               {complexContent}
            </TabsContent>
         );
         
         expect(screen.getByText('Tab 1 Header')).toBeInTheDocument();
         expect(screen.getByText('Tab 1 content')).toBeInTheDocument();
         expect(screen.queryByPlaceholderText('Form input')).not.toBeInTheDocument();
      });
   });

   describe('Snapshots', () => {
      it('matches snapshot with default props', () => {
         const { container } = render(<TabsContent {...defaultProps} />);
         
         expect(container.firstChild).toMatchSnapshot();
      });

      it('matches snapshot with useNewButton enabled', () => {
         const { container } = render(
            <TabsContent 
               {...defaultProps}
               useNewButton={true}
               newContent={<div>New tab content</div>}
            />
         );
         
         expect(container.firstChild).toMatchSnapshot();
      });

      it('matches snapshot with custom className', () => {
         const { container } = render(
            <TabsContent 
               {...defaultProps}
               className="custom-tabs-class"
            />
         );
         
         expect(container.firstChild).toMatchSnapshot();
      });

      it('matches snapshot with single child', () => {
         const { container } = render(
            <TabsContent 
               options={[{ label: 'Single', value: 'single' }]}>
               <div>Single content</div>
            </TabsContent>
         );
         
         expect(container.firstChild).toMatchSnapshot();
      });
   });
});
