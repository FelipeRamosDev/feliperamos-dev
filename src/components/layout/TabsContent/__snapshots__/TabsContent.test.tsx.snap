// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TabsContent Snapshots matches snapshot with custom className 1`] = `
<div
  class="mocked-parse-css"
>
  <div
    class="mocked-tabs-class"
  >
    <div
      data-testid="mui-tabs"
      data-value="0"
    >
      <button
        data-testid="mui-tab"
        type="button"
      >
        Tab 1
      </button>
      <button
        data-testid="mui-tab"
        type="button"
      >
        Tab 2
      </button>
      <button
        data-testid="mui-tab"
        type="button"
      >
        Tab 3
      </button>
    </div>
  </div>
  <div
    class="mocked-tab-content-class"
  >
    <div>
      Content 1
    </div>
  </div>
</div>
`;

exports[`TabsContent Snapshots matches snapshot with default props 1`] = `
<div
  class="mocked-parse-css"
>
  <div
    class="mocked-tabs-class"
  >
    <div
      data-testid="mui-tabs"
      data-value="0"
    >
      <button
        data-testid="mui-tab"
        type="button"
      >
        Tab 1
      </button>
      <button
        data-testid="mui-tab"
        type="button"
      >
        Tab 2
      </button>
      <button
        data-testid="mui-tab"
        type="button"
      >
        Tab 3
      </button>
    </div>
  </div>
  <div
    class="mocked-tab-content-class"
  >
    <div>
      Content 1
    </div>
  </div>
</div>
`;

exports[`TabsContent Snapshots matches snapshot with single child 1`] = `
<div
  class="mocked-parse-css"
>
  <div
    class="mocked-tabs-class"
  >
    <div
      data-testid="mui-tabs"
      data-value="0"
    >
      <button
        data-testid="mui-tab"
        type="button"
      >
        Single
      </button>
    </div>
  </div>
  <div
    class="mocked-tab-content-class"
  >
    <div>
      Single content
    </div>
  </div>
</div>
`;

exports[`TabsContent Snapshots matches snapshot with useNewButton enabled 1`] = `
<div
  class="mocked-parse-css"
>
  <div
    class="mocked-tabs-class"
  >
    <div
      data-testid="mui-tabs"
      data-value="0"
    >
      <button
        data-testid="mui-tab"
        type="button"
      >
        Tab 1
      </button>
      <button
        data-testid="mui-tab"
        type="button"
      >
        Tab 2
      </button>
      <button
        data-testid="mui-tab"
        type="button"
      >
        Tab 3
      </button>
      <button
        data-testid="mui-tab"
        type="button"
      >
        New
      </button>
    </div>
  </div>
  <div
    class="mocked-tab-content-class"
  >
    <div>
      Content 1
    </div>
  </div>
</div>
`;
